BangazonAPI/BangazonAPI/Controllers/PaymentTypeController.cs

```diff
@@ -0,0 +1,236 @@
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            AcctNumber = reader.GetString(reader.GetOrdinal("AcctNumber")),
+                            Name = reader.GetString(reader.GetOrdinal("Name")),
+                            CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId"))
+
+                        };
+
+                        paymenttypes.Add(paymenttype);
+                    }
+                    reader.Close();
+
+                    return Ok(paymenttypes);
+                }
+            }
+        }
+
+        //gets a single payment type
+
+        // GET api/<PaymentTypeController>/5
+        
+        [HttpGet("{id}", Name = "GetPaymentType")]
+        public async Task<IActionResult> Get([FromRoute] int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                        SELECT
+                            Id, AcctNumber, Name, CustomerId
+                        FROM PaymentType
+                        WHERE Id = @id";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    PaymentType paymenttype = null;
+
+                    if (reader.Read())
+                    {
+                        paymenttype = new PaymentType
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            AcctNumber = reader.GetString(reader.GetOrdinal("AcctNumber")),
+                            Name = reader.GetString(reader.GetOrdinal("Name")),
+                            CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId"))
+                        };
+                    }
+                    reader.Close();
+
+                    return Ok(paymenttype);
+                }
+            }
+        }
+
+        //posts a new payment type
+
+        // POST api/<PaymentTypeController>
+        [HttpPost]
+
+        public async Task<IActionResult> Post([FromBody] PaymentType paymenttype)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"INSERT INTO PaymentType (AcctNumber, Name, CustomerId)
+                                        OUTPUT INSERTED.Id
+                                        VALUES (@AcctNumber, @Name, @CustomerId)";
+                    cmd.Parameters.Add(new SqlParameter("@AcctNumber", paymenttype.AcctNumber));
+                    cmd.Parameters.Add(new SqlParameter("@Name", paymenttype.Name));
+                    cmd.Parameters.Add(new SqlParameter("@CustomerId", paymenttype.CustomerId));
```

Comment:
This is not necessarily something to change right now, but might be something to consider for the future. If I try to `POST` a payment type with a customer id that exists in the database, this works beautifully. If I try to `POST` a payment type with a customer Id that _doesn't_ exist in the database, I get a pretty ugly SQL Exception. 

Part of having your app "fail gracefully" involves thinking what's possible. For example, in the current iteration of this API it's totally possible to just enter a random number for `customerId`, which does break it, but if you knew for a fact that this API were going to be used with a UI that would _not_ allow that to happen, you might not have to worry about it. If for any reason you're unsure, and the user miiiight be able to break it this way, it would probably be worth going back in and adding a condition to check and see if the customer exists before you try to create the payment type. 

Just food for thought for later, not an immediate change.